x-pack/plugins/rule_registry/server/search_strategy/search_strategy.ts

```diff
@@ -47,10 +49,13 @@ export const ruleRegistrySearchStrategyProvider = (
           getActiveSpace(),
           alerting.getAlertingAuthorizationWithRequest(deps.request),
         ]);
-        const authzFilter = (await getAuthzFilter(
-          authorization,
-          ReadOperations.Find
-        )) as estypes.QueryDslQueryContainer;
+        let authzFilter;
+        if (!request.featureIds.some((featureId) => FEATURES_TO_SKIP_RBAC.includes(featureId))) {
```

Comment:
Thinking about it, if you have a mix of security solution and o11y, we should use RBAC but if you only have security solution do not use RBAC.

Reply:
IMO, we shouldn't support searching across security solution and o11y alerts until we're using Alerts RBAC for security solution alerts. The security solution team found that Alerts RBAC was insufficient for their use-case and wasn't granular enough, as a result we can't safely use it here.

Reply:
Should we reject requests that contain more than one feature id and one of those is SIEM?

Reply:
I think so... Lemme look more closely at the code and how it's used here, I'll let you know in a bit.

Reply:
My thinking was when we want to show alerts from security solutions and o11y then we should just go with RBAC and that's our way of doing it but security solution can just have it their own way for now. I still think it will be important to show correlation between o11y and security solution alerts

Reply:
Okay so the new logic is we reject requests that contain more than one feature id and one of those feature ids is siem. If it's just siem, we use the request user. All other requests use the internal user